### PR TITLE
Add MenuCommandService

### PR DESCRIPTION
**Pull request by AArnott**
## Proposed changes

- Add `MenuCommandService` and a few dependency types that are already identified as missing yet required for .NET 5.

## Test methodology 

- No tests have been ported (yet)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2650)

### CODE REVIEW

File: src/System.Design/src/ApiCompatBaseline.netcoreapp.txt [hunk 1]
```diff
@@ -7,14 +7,7 @@ TypesMustExist : Type 'System.ComponentModel.Design.ByteViewer' does not exist i
 TypesMustExist : Type 'System.ComponentModel.Design.ComponentActionsType' does not exist in the implementation but it does exist in the contract.
 TypesMustExist : Type 'System.ComponentModel.Design.DateTimeEditor' does not exist in the implementation but it does exist in the contract.
 TypesMustExist : Type 'System.ComponentModel.Design.DesignerActionListsChangedEventArgs' does not exist in the implementation but it does exist in the contract.
-TypesMustExist : Type 'System.ComponentModel.Design.DesignerActionListsChangedEventHandler' does not exist in the implementation but it does exist in the contract.
```
Comment (AArnott) [inline src/System.Design/src/ApiCompatBaseline.netcoreapp.txt]: The changes to this file were manually made. Running the `build -rebuild /p:BaselineAllAPICompatError=true` command as described in #2092 did not seem to update the file.

File: src/System.Design/src/System.Design.Forwards.cs [hunk 2]
```diff
@@ -40,3 +40,23 @@
...
+[assembly: TypeForwardedTo(typeof(System.ComponentModel.Design.DesignerCommandSet))]
+[assembly: TypeForwardedTo(typeof(System.ComponentModel.Design.MenuCommandsChangedEventArgs))]
+[assembly: TypeForwardedTo(typeof(System.ComponentModel.Design.MenuCommandsChangedEventHandler))]
+[assembly: TypeForwardedTo(typeof(System.ComponentModel.Design.MenuCommandsChangedType))]
+[assembly: TypeForwardedTo(typeof(System.ComponentModel.Design.MenuCommandService))]
+
+// TODO: This (and many more?!) were already defined in .NET Core but type forwarders did not exist,
+//       so anyone who compiled against netfx surface area would fail at runtime.
+//       How can we exhaustively list the types that are required?
+[assembly: TypeForwardedTo(typeof(System.ComponentModel.Design.DesignerActionItem))]
```
Comment (AArnott) [inline src/System.Design/src/System.Design.Forwards.cs]: I'm really interested in discussing the completeness of these type forwarders.

File: src/System.Design/src/ApiCompatBaseline.netcoreapp.txt (hunk 1 above)
Comment (RussKie) [reply src/System.Design/src/ApiCompatBaseline.netcoreapp.txt]: We haven't implemented API compat for 3.1/5.0 yet, so it won't work

File: src/System.Design/src/System.Design.Forwards.cs (hunk 2 above)
Comment (RussKie) [reply src/System.Design/src/System.Design.Forwards.cs]: you only need to TF `internal` types, all `public` types are TF'ed automagically via the facade assembly
you can confirm this via ILSpy

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandService.cs [hunk 3]
```diff
@@ -0,0 +1,634 @@
...
+            DocumentManager docManager = (DocumentManager)serviceProvider.GetService(typeof(DocumentManager));
+            if (docManager != null) {
+                if (docManager.ActiveDesigner != null) {
+                    ISelectionService svc = (ISelectionService)docManager.ActiveDesigner.GetService(typeof(ISelectionService));
+                    if (svc != null) {
+                        docManager.OnSelectionChanged(svc);
+                    }
+                }
+            }
+            */
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandService.cs]: We don't need this, please remove

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandService.cs [hunk 4]
```diff
@@ -0,0 +1,634 @@
...
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandService.cs]: ditto

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandsChangedEventArgs.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+namespace System.ComponentModel.Design
+{
+    /// <summary>
+    ///     This EventArgs class is used by the MenuCommandService to signify
+    ///     that there has been a change in MenuCommands (added or removed)
+    ///     on the related object.
+    /// </summary>
+    [System.Runtime.InteropServices.ComVisible(true)]
+    public class MenuCommandsChangedEventArgs : EventArgs
```
Comment (RussKie) [inline src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandsChangedEventArgs.cs]: Please clean up the file, remove `this`, simplify properties

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandService.cs (hunk 3 above)
Comment (AArnott) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandService.cs]: Done

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandService.cs (hunk 4 above)
Comment (AArnott) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandService.cs]: Done

File: src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandsChangedEventArgs.cs (hunk 5 above)
Comment (AArnott) [reply src/System.Windows.Forms.Design/src/System/ComponentModel/Design/MenuCommandsChangedEventArgs.cs]: Done

**Pull request merged**